src/wazuh_modules/syscollector/tests/sysCollectorFlatbuffers/syscollectorFb_test.cpp

```diff
@@ -0,0 +1,286 @@
+                    builder,
+                    architecture,
+                    checksum,
+                    description,
+                    format,
+                    install_time,
+                    groups,
+                    item_id,
+                    multiarch,
+                    name,
+                    location,
+                    priority,
+                    scan_time,
+                    size_attr,
+                    source,
+                    vendor,
+                    version);
+
+    auto index = builder.CreateString("00");
+    auto timestamp = builder.CreateString("00/00/0000T00:00:00.000");
+
+    auto packageData = CreatePackageData(builder, data, index, timestamp);
+
+    auto sync_msg = CreateSyncMsg(builder, component, Data_PackageData, packageData.Union(), type);
+
+    builder.Finish(sync_msg);
+
+    uint8_t* buf = builder.GetBufferPointer();
+    int size = builder.GetSize();
+
+    std::ofstream ofile(SYNC_PMSG_TEST_PATH, std::ios::binary);
+    ofile.write((char*)buf, size);
+    ofile.close();
+}
+
+TEST(SyscollectorFbTest, readBinaryPackageUnix)
+{
+    std::ifstream infile;
+    infile.open(SYNC_PMSG_TEST_PATH, std::ios::binary | std::ios::in);
+    infile.seekg(0, std::ios::end);
+    int length = infile.tellg();
+    infile.seekg(0, std::ios::beg);
+    char* raw_data = new char[length];
+    infile.read(raw_data, length);
+    infile.close();
+
+    auto sync_msg = GetSyncMsg(raw_data);
+
+    EXPECT_STREQ(sync_msg->component()->c_str(), COMPONENT);
+    EXPECT_STREQ(sync_msg->type()->c_str(), TYPE);
+    EXPECT_EQ(sync_msg->data_type(), Data_PackageData);
+
+    auto data = static_cast<const PackageData*>(sync_msg->data());
+
+    EXPECT_STREQ(data->index()->c_str(), "00");
+    EXPECT_STREQ(data->timestamp()->c_str(), "00/00/0000T00:00:00.000");
+    EXPECT_STREQ(data->data()->architecture()->c_str(), "x86");
+    EXPECT_STREQ(data->data()->checksum()->c_str(), "md5sha1md5sha1");
+    EXPECT_STREQ(data->data()->description()->c_str(), "Nice");
+    EXPECT_STREQ(data->data()->format()->c_str(), "UTF-8");
+    EXPECT_STREQ(data->data()->install_time()->c_str(), "");
+    EXPECT_STREQ(data->data()->groups()->c_str(), "wazuh");
+    EXPECT_STREQ(data->data()->item_id()->c_str(), "00");
+    EXPECT_STREQ(data->data()->multiarch()->c_str(), "yes");
+    EXPECT_STREQ(data->data()->name()->c_str(), "NicePackage");
+    EXPECT_STREQ(data->data()->location()->c_str(), "\\home");
+    EXPECT_STREQ(data->data()->priority()->c_str(), "01");
+    EXPECT_STREQ(data->data()->scan_time()->c_str(), "00/00/0000T00:00:00.000");
+    EXPECT_EQ(data->data()->size(), 1);
+    EXPECT_STREQ(data->data()->source()->c_str(), "wazuh");
+    EXPECT_STREQ(data->data()->vendor()->c_str(), "wazuh");
+    EXPECT_STREQ(data->data()->version()->c_str(), "v1");
+
+    std::remove(SYNC_PMSG_TEST_PATH);
+    delete []raw_data;
+}
+
+TEST(SyscollectorFbTest, createBinaryPackageWindows)
```

Comment:
This test doesn't do any assertion. It only creates the file that is used in the test "readBinaryPackageWindows". So, this test could be merged with "readBinaryPackageWindows" in a unique test.